tests/comparisons/Controller/testBakeActionsContent.php

```diff
@@ -13,9 +15,9 @@ class BakeArticlesController extends AppController
     /**
      * Index method
      *
-     * @return \Cake\Http\Response|null|void Renders view
+     * @return void Renders view
      */
-    public function index()
+    public function index(): void
```

Comment:
I think we avoided adding hints to these methods before because `?Response` doesn't include null and there isn't a way to have a union between a type and `void`.

Should we these `void` actions start ending with `return $this->render()` and add the `?Response` return type to make it more obvious what is going on and set us up for a smoother upgrade in the future?

Reply:
Indeed `void` isn't allowed be part of union return type (currently). Therefore we can't do `?Response|void`
I personally would prefer adding the `return $this->render();` at the end of each generated controller action (except `delete` of course) to enable us to have `?Response` for `add` and `edit` as well as just `Response` for `index` and `view` as a return type.
Users have the ability right now to do so - it just depends on if we want to enforce that in the future or not.

Reply:
I personally would avoid setting the types here.
It is not uncommon to have to redirect in those actions (permissions, ...), and then you need to completely rewrite the types here.
It is easier IMO to leave the range open.

Reply:
`return $this->render();` sounds like quite some noise and overhead for no real value gain?

Reply:
I would simply leave out the explicit type instead of adding `void` as is doesn't add much value. There's no app code which calls action method and Controller::invokeAction() already throws an exception if the method reruns anything besides ResponseInterface (or null).

Reply:
My intention here was more like making the auto-generated code base a bit more uniform with how `5.x` has been treated till now. Especially since we just added return types and property types to basically everything in `5.x` it just looks weird to me to not have it for controller actions as well.

I do though understand that it doesn't really provide a benefit since (as you already said) controller actions don't get called by any other parts of the codebase beside `Controller::invokeAction()`.

Reply:
Controller and command code, both.
They are the main glue of the layers and especially here it can be useful to allow a more lean approach in dev.
At least thats how we were used to it so far :)